bootstrap/viaproxy/src/main/java/org/geysermc/geyser/platform/viaproxy/GeyserViaProxyBootstrap.java

```diff
@@ -0,0 +1,161 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+package org.geysermc.geyser.platform.viaproxy;
+
+import net.raphimc.vialegacy.api.LegacyProtocolVersion;
+import net.raphimc.viaproxy.cli.options.Options;
+import org.apache.logging.log4j.Logger;
+import org.geysermc.geyser.GeyserBootstrap;
+import org.geysermc.geyser.GeyserImpl;
+import org.geysermc.geyser.GeyserLogger;
+import org.geysermc.geyser.api.network.AuthType;
+import org.geysermc.geyser.api.util.PlatformType;
+import org.geysermc.geyser.command.GeyserCommandManager;
+import org.geysermc.geyser.configuration.GeyserConfiguration;
+import org.geysermc.geyser.dump.BootstrapDumpInfo;
+import org.geysermc.geyser.ping.GeyserLegacyPingPassthrough;
+import org.geysermc.geyser.ping.IGeyserPingPassthrough;
+import org.geysermc.geyser.text.GeyserLocale;
+import org.geysermc.geyser.util.FileUtils;
+import org.geysermc.geyser.util.LoopbackUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+
+public class GeyserViaProxyBootstrap implements GeyserBootstrap {
+
+    private final File rootFolder;
+    private final GeyserViaProxyLogger logger;
+    private GeyserViaProxyConfiguration config;
+
+    private GeyserImpl geyser;
+    private GeyserCommandManager commandManager;
+    private IGeyserPingPassthrough pingPassthrough;
+
+    public GeyserViaProxyBootstrap(final Logger logger, final File rootFolder) {
+        this.logger = new GeyserViaProxyLogger(logger);
+        this.rootFolder = rootFolder;
+    }
+
+    @Override
+    public void onGeyserInitialize() {
+        try {
+            final File configFile = FileUtils.fileOrCopiedFromResource(new File(this.rootFolder, "config.yml"), "config.yml", s -> s.replaceAll("generateduuid", UUID.randomUUID().toString()), this);
+            this.config = FileUtils.loadConfig(configFile, GeyserViaProxyConfiguration.class);
+        } catch (IOException e) {
+            this.logger.severe(GeyserLocale.getLocaleStringLog("geyser.config.failed"), e);
+            return;
+        }
```

Comment:
After loading the config, you should set the debug level of the geyser logger - that's why it's usually created after loading the config. The approach here is fine though, as long as you set debug 
<https://github.com/GeyserMC/Geyser/blob/master/bootstrap%2Fvelocity%2Fsrc%2Fmain%2Fjava%2Forg%2Fgeysermc%2Fgeyser%2Fplatform%2Fvelocity%2FGeyserVelocityPlugin.java#L104>